api/src/main/java/org/openmrs/module/fhir2/api/search/param/BaseResourceSearchParams.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.search.param;
+
+import java.io.Serializable;
+import java.util.HashSet;
+
+import ca.uhn.fhir.model.api.Include;
+import ca.uhn.fhir.rest.api.SortSpec;
+import ca.uhn.fhir.rest.param.DateRangeParam;
+import ca.uhn.fhir.rest.param.TokenAndListParam;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import lombok.experimental.SuperBuilder;
+
+/**
+ * Class containing shared search (and search result) parameters that apply to all resources See
+ * https://www.hl7.org/fhir/search.html#Summary for a comprehensive description of these parameters
+ */
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@SuperBuilder
+public class BaseResourceSearchParams implements Serializable {
```

Comment:
did a first pass at pulling out params that are used in most resources, although it does seem like resources explicitly define which params they accept within the resource Provider class directly. wondering if this means that not all resources will support the common parameters defined in https://www.hl7.org/fhir/search.html#Summary ?

Reply:
also technically we have common search params and common search result params, not sure if it's better to distinguish between those two and have BaseResourceSearchParams/BaseResourceSearchResultParams

Reply:
Hmmmm... yeah. It might make sense to implement some common parameters, but I'd want to discuss how we do it. The only two common search parameters that are unproblematically acceptable for all types are `_id` and `_lastUpdated`.

Reply:
are you ok with me leaving the common params as is for now? sounds like we'll at least have _id and _lastUpdated in that class, so imo it can stick around even if we pull out some params that are currently defined there in the future.